druid/src/window.rs

```diff
@@ -339,6 +416,21 @@ impl<T: Data> Window<T> {
             env,
             Rect::from_origin_size(Point::ORIGIN, size),
         );
+
+        for modal in &mut self.modals {
+            let bc = BoxConstraints::new(Size::ZERO, size);
+            let modal_size = modal.widget.layout(&mut layout_ctx, &bc, data, env);
+            let modal_origin = if let Some(pos) = modal.position {
+                // TODO: translate the position to ensure that the modal fits in our bounds.
+                pos
+            } else {
+                ((size.to_vec2() - modal_size.to_vec2()) / 2.0).to_point()
```

Comment:
I believe these numbers should be rounded so they land on pixels and not in-between them.